### DISCUSSION TITLE
Internal order of the constraint multipliers

### DISCUSSION THREAD

**Discussion by 309815765** [coin-or/Ipopt#502] [Q&A] [answered]
Hi,

I need to get the constraint multipliers for sensitivity analysis. To my knowledge, the constraint with both lower bound and upper bound will be converted to two constraints (correct me if it's wrong). I'd like to know how does Ipopt order multipliers (lambda) for constraints, especially for the constraints with both boundaries.

Thanks

**Comment by svigerske** [comment, marked as answer]
If lower and upper bounds are equal, then the constraint will be converted from `lhs <= g(x) <= rhs` to `g(x) - lhs = 0`.   
If lower and upper bounds are different (and possibly infinite), then the constraint will be converted from `lhs <= g(x) <= rhs` to `g(x) - s = 0`, `lhs <= s <= rhs`, where s is a slack variable that gets introduced.
See also https://coin-or.github.io/Ipopt/classIpopt_1_1TNLPAdapter.html#details

The bound multiplier lambda for constraints is taken from the bound multiplier of  `g(x) - lhs = 0` (if lhs==rhs) or `g(x) - s = 0` (if lhs != rhs).
So in the latter case, the sign of lambda tells you whether `lhs <= g(x)` or `g(x) <= rhs` is active.

**Reply by 309815765** [reply]
Thanks for your response!

So, `lambda[i]` is the multiplier for `g[i]`, and `lamdda[i] > 0` indicates that `g[i] <= rhs` is active whereas `lambda[i] < 0` means `lhs<= g[i]` is active.

**Reply by svigerske** [reply]
Or the other way around. I cannot remember signs.
